lib/collection/src/collection_manager/search_result_aggregator.rs

```diff
@@ -58,16 +58,14 @@ impl BatchResultAggregator {
         }
     }
 
-    pub fn update_point_versions(&mut self, search_results: &Vec<Vec<Vec<ScoredPoint>>>) {
-        for segment_result in search_results {
-            for segment_batch_result in segment_result {
-                for point in segment_batch_result {
-                    let point_id = point.id;
-                    let point_version =
-                        self.point_versions.entry(point_id).or_insert(point.version);
-                    *point_version = max(*point_version, point.version);
-                }
-            }
+    pub fn update_point_versions<'a>(
+        &mut self,
+        all_searches_results: impl Iterator<Item = &'a ScoredPoint>,
+    ) {
+        for point in all_searches_results {
+            let point_id = point.id;
+            let point_version = self.point_versions.entry(point_id).or_insert(point.version);
+            *point_version = max(*point_version, point.version);
```

Comment:
Potential optimization

```suggestion
            self.point_versions
                .entry(point_id)
                .and_modify(|point_version| {
                    *point_version = max(*point_version, point.version);
                })
                .or_insert(point.version);
```